### DISCUSSION TITLE
Could not automatically save pickle file

### DISCUSSION THREAD

**Discussion by leelew** [MilesCranmer/PySR#708] [Q&A] [answered]
Hi,

After fitting data by PySRRegressor, I only have two saved files, .csv and .bkup. However, I want to reload all equations after fitting, but I cannot apply  `from_file` because .pkl file is absent. Is there any suggestion?
Thanks!

Regards,
Lu Li

Code is attached:

model = PySRRegressor(
    #procs=10,
    populations=3*48,
    ncycles_per_iteration=5000,
    niterations=1000000000,  # Run forever
    timeout_in_seconds=int(3600*n_hour),
    maxsize=50,
    complexity_of_operators = {
                            **{key: 3*verylow_complexity for key in medium_complex_ops},
                            **{key: 2*verylow_complexity for key in low_complex_ops},
                            **{key: 1*verylow_complexity for key in very_low_complex_ops}},
    complexity_of_variables=1,
    binary_operators=binary_operators,
    unary_operators=unary_operators,
    extra_sympy_mappings={'inv': lambda x: 1/x},
    model_selection="best",
    constraints={"pow": (9, 1)},
    nested_constraints={"tanh": {"tanh":0}},
    loss="loss(x, y) = (abs(max(min(x,100),0) - y))^%d"%loss_exp, #(julia syntax)
    temp_equation_file=True,
    delete_tempfiles=False,
    #cluster_manager="slurm",
    #parsimony=10,
    bumper=True,
    weight_optimize=0.001,)

**Comment by leelew** [comment, marked as answer]
Hi,

I found `from_file` could directly load equations from .csv file. Thus my problem is solved.

I have a further question. I want to count free parameters of equations instead of directly use complexity. Is there any method to automatically get the number of free parameters? I have to manually count it for all equations after fitting data.

Thanks!

Regards,
Lu Li

**Reply by MilesCranmer** [reply]
Want to make a new thread for the constants?

And great to hear you could fix it with the .csv file loading!

**Reply by leelew** [reply]
I try to count constant in the resulting equations using `sympy` library. However, I did not know how to count it. e.g., $1/(ax+b)$. Is there any suggestions?

**Reply by leelew** [reply]
I found use `sympy` could count constants directly by following codes:

```python
import sympy as sp
for i in range(len(model.equations_)):
    expr = model.equations_.sympy_format.iloc[i]
    print(expr)
    constants = [term for term in expr.atoms(sp.Number)]
    constants = np.setdiff1d(constants, [-1])
    print(len(constants))
